### ISSUE TITLE
🐛 BUG: Hoisted scripts do not evaluate `src` expression

### ISSUE THREAD

**Issue by loilo** [snowpackjs/astro#1320]
### What package manager are you using?

npm, yarn

### What operating system are you using?

macOS

### Describe the Bug

Hoisted external scripts don't work as described [in the docs](https://docs.astro.build/core-concepts/astro-components#hoisted-scripts):

```jsx
<script src={Astro.resolve('./my-component.js')} hoist></script>
```

The compiler does not evaluate expressions passed to a hoisted script `src`, it will instead process it as a bare string:

https://github.com/snowpackjs/astro/blob/e686c3c50469d18db93da7ce79ddcac8659c3166/packages/astro/src/compiler/codegen/index.ts#L678-L680

This behavior has not been caught because it's currently not covered by [the according test](https://github.com/snowpackjs/astro/blob/e686c3c50469d18db93da7ce79ddcac8659c3166/packages/astro/test/astro-scripts.test.js#L12-L21).

### Steps to Reproduce

1. `npm init astro` using the "minimal" template
2. Create a `main.js`:

   ```sh
   echo 'console.log("hello world") > src/main.js'
   ```
3. Insert `<script src={Astro.resolve('../js/main.js')} hoist></script>` somewhere into the `src/pages/index.astro`.

   ```sh
   sed -i '' 's|<body>|<body>\n\t<script src={Astro.resolve("../js/main.js")} hoist></script>|' src/pages/index.astro
   ```
4. Build project.

   ```sh
   npm run build
   ```
5. The build failed, because it could not find a script with the path `/Astro.resolve(...`. As per the docs, the expression should be properly resolved instead.

   ```
   [build] Could not load /Astro.resolve( (imported by astro-virtual:/index.js): ENOENT: no such file or directory, open '/Astro.resolve('
   npm ERR! code ELIFECYCLE
   npm ERR! errno 1
   npm ERR! @example/minimal@0.0.1 build: `astro build`
   npm ERR! Exit status 1
   npm ERR!
   npm ERR! Failed at the @example/minimal@0.0.1 build script.
   npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
   
   npm ERR! A complete log of this run can be found in:
   npm ERR!     /Users/florian/.npm/_logs/2021-09-05T17_34_33_343Z-debug.log
   ```

**Comment by jasikpark** [comment]
I agree, that should work!

**Comment by jasikpark** [comment]
Actually, these two features honestly work best together! This definitely needs to be fixed soon..

I've got a successfully failing reproduction of the bug here: https://codesandbox.io/s/astro-1320-reproduction-4criv?file=/src/pages/index.astro

**Comment by jasikpark** [comment]
Created a PR to work on fixing this: #1336. I'm honestly not sure how to translate the attribute into the finished resolved URL, but maybe @matthewp knows!

**Comment by FredKSchott** [comment]
@jasikpark worked on a failing test that illustrates this issue here! https://github.com/jasikpark/astro/tree/fix-hoist-plus-astro-resolve

Would love for someone to take a look and getting the test passing, if they wanted to fix this bug!

**Comment by akellbl4** [comment]
> For this issue, I would recommend waiting until `Astro.resolve` support is added in the `next` branch.

https://github.com/snowpackjs/astro/pull/1405#issuecomment-928277131

**Comment by natemoo-re** [comment]
Update: `Astro.resolve` support has been added to `next` and `next` has been promoted to `main`. However, we still have an issue here (at least [in this repro](https://stackblitz.com/edit/github-gqnt8g?file=src%2Fpages%2Findex.astro)). We'll keep this open until everything is working as expected!

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
